### PR TITLE
IBM-Swift/Kitura#646 - Added entry/exit logging

### PR DESCRIPTION
**Pull request by rob-deans**

### CODE REVIEW

**Comment by ianpartridge** [comment]
@shmuelk @rfdickerson can you review - are you happy with this approach?

**Comment by shmuelk** [comment]
I think entry and exit need to have the lowest rawValues. The filtering code does by rawValue, the order is important.

I also think that it's important to add a function to test if a log Level is enabled, especially with regard to entry and exit logging. The performance hit of creating the message can sometimes be more than you would think. The function signature should look something like this:

  isLogging(level: LoggerMessageType) -> Bool

File: Sources/LoggerAPI/Logger.swift [hunk 1]
```diff
@@ -35,6 +37,10 @@ extension LoggerMessageType: CustomStringConvertible {
             return "WARNING"
         case .error:
             return "ERROR"
+        case .entry:
```
Comment (ianpartridge) [inline Sources/LoggerAPI/Logger.swift]: For consistency, the case statements should go in the same order as the enum.

**Comment by rob-deans** [comment]
@shmuelk Ready for another review, thanks for your comments

**Comment by shmuelk** [comment]
Looks good now. merging.

**Pull request merged**